tests/test_dependency_rendering.py

```diff
@@ -491,3 +491,131 @@ def test_multiple_components_all_placeholders_removed(self):
         template = Template(template_str)
         rendered = create_and_process_template_response(template)
         self.assertNotIn("_RENDERED", rendered)
+
+    def test_adds_component_id_html_attr_single(self):
+        registry.register(name="test", component=SimpleComponent)
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component 'test' variable='foo' / %}
+        """
+        template = Template(template_str)
+        rendered = create_and_process_template_response(template)
+
+        self.assertHTMLEqual(rendered, "Variable: <strong data-djc-id-a1bc3f>foo</strong>")
+
+    def test_adds_component_id_html_attr_single_multiroot(self):
+        class SimpleMultiroot(SimpleComponent):
+            template: types.django_html = """
+                Variable: <strong>{{ variable }}</strong>
+                Variable2: <div>{{ variable }}</div>
+                Variable3: <span>{{ variable }}</span>
+            """
+
+        registry.register(name="test", component=SimpleMultiroot)
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component 'test' variable='foo' / %}
+        """
+        template = Template(template_str)
+        rendered = create_and_process_template_response(template)
+
+        self.assertHTMLEqual(
+            rendered,
+            """
+            Variable: <strong data-djc-id-a1bc3f>foo</strong>
+            Variable2: <div data-djc-id-a1bc3f>foo</div>
+            Variable3: <span data-djc-id-a1bc3f>foo</span>
+            """,
+        )
+
+    # Test that, if multiple components share the same root HTML elements,
+    # then those elemens will have the `data-djc-id-` attribute added for each component.
+    def test_adds_component_id_html_attr_nested(self):
+        class SimpleMultiroot(SimpleComponent):
+            template: types.django_html = """
+                Variable: <strong>{{ variable }}</strong>
+                Variable2: <div>{{ variable }}</div>
+                Variable3: <span>{{ variable }}</span>
+            """
+
+        class SimpleOuter(SimpleComponent):
+            template: types.django_html = """
+                {% load component_tags %}
+                {% component 'multiroot' variable='foo' / %}
+                <div>Another</div>
+            """
+
+        registry.register(name="multiroot", component=SimpleMultiroot)
+        registry.register(name="outer", component=SimpleOuter)
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component 'outer' variable='foo' / %}
+        """
+        template = Template(template_str)
+        rendered = create_and_process_template_response(template)
+
+        self.assertHTMLEqual(
+            rendered,
+            """
+            Variable: <strong data-djc-id-a1bc3f data-djc-id-a1bc41>foo</strong>
```

Comment:
Here's an interesting thing.

Vue also uses component IDs format like `data-djc-id-a1bc3f`, so as boolean attributes with no value

At some point, before I realized why, I tried to set the component IDs as key=value attributs, e.g. `data-djc-id="a1bc3f"`.

But the problem with that is that a single HTML element may be a root element for multiple components.

E.g. consider this:

```py
@register("inner")
class Inner(Component):
    template = """
    <div>123</div>
    """

class Outer(Component):
    template = """
    {% component "inner" %}
    """
```

In this case, the `<div>123</div>` will be the root element for both the `Outer` and `Inner` components.

But if the component ID was set as key=value pair, they would overwrite each other. Because the browser parses:

```html
<div data-djc-id="a1bc3f" data-djc-id="d3d3d3">123</div>
```

as:

```html
<div data-djc-id="d3d3d3">123</div>
```

Now, on the other hand, if we define the IDs as boolean attributes, we can have as many of them without fearing that they would overwrite one-another:

```html
<div data-djc-id-a1bc3f data-djc-id-d3d3d3>123</div>
```